### PR TITLE
feat: add support for observation point ini file

### LINKED ISSUE #160: Support ObsFile for observation stations
As referenced in the MDU/```FMModel``` under ```[output] ObsFile```.
Only support the "new style" ```*_obs.ini```, as documented in Sec. F.2.2.1: https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#subsubsection.F.2.2.1

Make sure to add validators + tests that check the optional requiredness of branchId + chainage, or x+y.

### PR DESCRIPTION
**Pull request by arthurvd**
Fixes #160.

### CODE REVIEW

File: hydrolib/core/io/obs/models.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    name: str = Field("id", max_length=255, alias="name")
+    locationtype: Optional[LocationType] = Field(None, alias="locationType")
+
+    branchid: Optional[str] = Field(None, alias="branchId")
+    chainage: Optional[float] = Field(None, alias="chainage")
+
+    x: Optional[float] = Field(None, alias="x")
+    y: Optional[float] = Field(None, alias="y")
+
```
Comment (priscavdsluis) [inline hydrolib/core/io/obs/models.py:68]: Should we add a validator for that one of the two options is required: branchId + chainage or x + y?

File: tests/io/test_obs.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        assert obspoint.locationtype == None
+        assert obspoint.branchid == obsvalues["branchid"]
+        assert obspoint.chainage == obsvalues["chainage"]
+
+    @pytest.mark.parametrize(
+        "loctype, expected_type",
+        [
+            ("1d", LocationType.oned),
+            ("2d", LocationType.twod),
+        ],
```
Comment (priscavdsluis) [inline tests/io/test_obs.py:37]: Add "all" to test cases?

File: hydrolib/core/io/obs/models.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            "y-coordinate of the location of the observation point.",
+            alias="y",
+        )
+
+    comments: Comments = Comments()
+
+    _header: Literal["ObservationPoint"] = "ObservationPoint"
+
+    name: str = Field("id", max_length=255, alias="name")
+    locationtype: Optional[LocationType] = Field(None, alias="locationType")
```
Comment (priscavdsluis) [inline hydrolib/core/io/obs/models.py:61]: Should there be a validator for the following? (from the manual -> locationType) : 

> Only when x and y are also specified. 1d: snap to closest 1D grid point, 2d: snap to closest 2D grid cell centre, all: snap to closest 1D or 2D point.

File: tests/data/input/dflowfm_individual_files/obsPoints_obs.ini [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+x            = -2.06250000e+00
+y            = 5.71583333e+01
+
+[ObservationPoint]
+name         = AVMH
+x            = -2.73750000e+00
+y            = 5.15083334e+01
+
+[ObservationPoint]
+name         = #St Helier Jersey#
```
Comment (priscavdsluis) [inline tests/data/input/dflowfm_individual_files/obsPoints_obs.ini:17]: Isn't something between two # a comment?

**Comment by priscavdsluis** [comment]
Do you know why this PR doesn't show up on SonarCloud?

File: tests/io/test_obs.py (hunk 2 above)
Comment (arthurvd) [reply tests/io/test_obs.py:37]: done

File: tests/data/input/dflowfm_individual_files/obsPoints_obs.ini (hunk 4 above)
Comment (arthurvd) [reply tests/data/input/dflowfm_individual_files/obsPoints_obs.ini:17]: No, this is our ugly Deltares-customized ini format: Single # is a comment char, whereas a strings enclosed in two #'s are string literal with possible whitespace. It is how it is... ;-)

**Comment by priscavdsluis** [comment]
Created isse #263 263

**Review by priscavdsluis — APPROVED** [review]

**Pull request merged**